TLM/TLM/UI/Helpers/DropDownOption.cs

```diff
@@ -0,0 +1,98 @@
+namespace TrafficManager.UI.Helpers {
+    using ICities;
+    using ColossalFramework.UI;
+    using TrafficManager.State;
+    using CSUtil.Commons;
+    using System;
+    using System.Linq;
+    using TrafficManager.API.Traffic.Enums;
+
+    public class DropDownOption<TEnum> : SerializableUIOptionBase<TEnum, UIDropDown, DropDownOption<TEnum>>
+        where TEnum : struct, Enum, IConvertible {
+
+        public DropDownOption(string fieldName, Options.PersistTo scope = Options.PersistTo.Savegame)
+        : base(fieldName, scope) { }
+
+        /* Data */
+        private static TEnum[] values_ = Enum.GetValues(typeof(TEnum)) as TEnum[];
+
+        private static string[] names_ = Enum.GetNames(typeof(TEnum));
+
+        private static string[] keys_ =
```

Comment:
Should these be instance based?

Reply:
No. why? 
The only advantage of instance is that it will cleanup memory when TMPE is disabled. is that what you want?

Reply:
I thought only the instance was typed (TEnum) - how does that apply to the static fields/methods?